raiden/network/transport/matrix/client.py

```diff
@@ -392,22 +394,24 @@ def _sync(self, timeout_ms=30000):
             self._post_hook_func(self.sync_token)
 
     def _handle_response(self, response, first_sync=False):
+        if not self.should_listen:
+            return
         # Handle presence after rooms
         for presence_update in response["presence"]["events"]:
-            for callback in self.presence_listeners.values():
+            for callback in list(self.presence_listeners.values()):
```

Comment:
Because during shutdown the `presence_listeners` dict gets modified which can lead to `dictionary changed size during iteration` exceptions when this happens while this loop is running.

This only started appearing now that the shutdown is handled properly.